src/shims/intrinsics.rs

```diff
@@ -33,6 +29,25 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         // that might still hang around!
 
         let intrinsic_name = &*tcx.item_name(instance.def_id()).as_str();
+
+        // The 'abort' intrinsic has an empty 'dest', so we check for it before
+        // unwrapping it
+        if intrinsic_name == "abort" {
+            // FIXME: Add a better way of indicating 'abnormal' termination,
+            // since this is not really an 'unsupported' behavior
+            throw_unsup_format!("the evaluated program aborted!");
+        }
+
+        if intrinsic_name == "miri_start_panic" {
+            return this.handle_miri_start_panic(args, unwind)
+        }
```

Comment:
Please make this a match, like we do for the others. We have a similar structure in `foreign_items`: first one match for things that diverge, then another for the rest.